src/fides/api/ops/models/policy.py

```diff
@@ -190,31 +198,39 @@ def _is_ancestor_of_contained_categories(
 
     TODO: Should we memoize this function?
     """
-    ref = _get_ref_from_taxonomy(fides_key=fides_key)
+    ref = _get_ref_from_taxonomy(
+        fides_key=fides_key,
+        all_categories=all_categories,
+    )
     if ref.parent_key:
         if ref.parent_key in data_categories:
             return True, ref.parent_key
 
         return _is_ancestor_of_contained_categories(
             fides_key=ref.parent_key,
             data_categories=data_categories,
+            all_categories=all_categories,
         )
 
     return False, None
 
 
-def _validate_rule_target_collection(target_categories: List[str]) -> None:
+def _validate_rule_target_collection(
+    db: Session,
+    target_categories: List[str],
+) -> None:
     """
     Validates that no erasure rules within the Policy have conflicting data category targets:
         - We cannot mask the same data categories multiple times
         - We cannot perform separate masking upon a data category's sub-categories
     """
-
+    all_categories = DataCategory.all(db=db)
```

Comment:
I've made a point of fetching these here because:
- a DB session is available from where we're calling this method
- this is the last point before entering a recursive flow